libtenzir/src/tql2/parser.cpp

```diff
@@ -237,6 +239,14 @@ class parser {
       auto entity = parse_entity();
       return parse_invocation(std::move(entity));
     }
+    // This is a hack: `move` is both a unary expression operator and a
+    // pipeliine operator. In order to support both, we need to special-case the
+    // pipeline operator parsing, as the tokenizer returns `tk::move` instead of
+    // `tk::identifier`.
+    if (silent_peek(tk::move)) {
+      auto entity = parse_entity(expect(tk::move).as_identifier());
+      return parse_invocation(std::move(entity));
+    }
```

Comment:
I couldn't find a better way to do this. It does work for `if`, but that's both a binary operator and a statement/operator, and I think the conflict here is between statements/operators and unary operators.

Reply:
Maybe contextual keywords is the proper solution here, but I think this is fine for now.

Reply:
I agree. I'll leave the comment in as-is, but will resolve this thread.